diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9a9b1db9a..1302243af 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -495,10 +495,30 @@ target_compile_definitions(
 if (WIN32)
   target_compile_definitions(
     freetype PRIVATE _CRT_SECURE_NO_WARNINGS _CRT_NONSTDC_NO_WARNINGS)
+  if(WINDOWS_STORE)
+    target_compile_definitions(freetype PRIVATE _WINRT_DLL)
+  endif()
   if (BUILD_SHARED_LIBS)
     target_compile_definitions(
       freetype PRIVATE DLL_EXPORT)
   endif ()
+  set_target_properties(freetype PROPERTIES
+    COMPILE_PDB_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}
+    COMPILE_PDB_NAME freetype
+    COMPILE_PDB_NAME_DEBUG freetyped
+    PDB_NAME freetype
+    PDB_NAME_DEBUG freetyped
+  )
+  target_compile_options(freetype
+    PRIVATE
+      /MP
+      /sdl-
+  )
+  target_link_options(freetype
+    PRIVATE
+      /INCREMENTAL:NO
+      /debug:full
+  )
 else ()
   # Windows handles exports with DLL_EXPORT defined above.
   set_target_properties(freetype PROPERTIES
@@ -655,6 +675,17 @@ if (NOT SKIP_INSTALL_LIBRARIES AND NOT SKIP_INSTALL_ALL)
     DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig
     COMPONENT pkgconfig)
 
+  install(
+    FILES ${PROJECT_BINARY_DIR}/RelWithDebInfo/freetype.pdb
+    DESTINATION lib
+    CONFIGURATIONS RelWithDebInfo
+  )
+  install(
+    FILES ${PROJECT_BINARY_DIR}/Debug/freetyped.pdb
+    DESTINATION lib
+    CONFIGURATIONS Debug
+  )
+
   include(CMakePackageConfigHelpers)
   write_basic_package_version_file(
     ${PROJECT_BINARY_DIR}/freetype-config-version.cmake
