diff --git a/CMakeLists.txt b/CMakeLists.txt
index ce70f6b5e..6113a79e9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -495,10 +495,30 @@ target_compile_definitions(
 if (WIN32)
   target_compile_definitions(
     freetype PRIVATE _CRT_SECURE_NO_WARNINGS _CRT_NONSTDC_NO_WARNINGS)
+  if(WINDOWS_STORE)
+    target_compile_definitions(freetype PRIVATE _WINRT_DLL)
+  endif()
   if (BUILD_SHARED_LIBS)
     target_compile_definitions(
       freetype PRIVATE DLL_EXPORT)
   endif ()
+  set_target_properties(freetype PROPERTIES
+    COMPILE_PDB_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}
+    COMPILE_PDB_NAME freetype
+    COMPILE_PDB_NAME_DEBUG freetyped
+    PDB_NAME freetype
+    PDB_NAME_DEBUG freetyped
+  )
+  target_compile_options(freetype
+    PRIVATE
+      /MP
+      /sdl-
+  )
+  target_link_options(freetype
+    PRIVATE
+      /INCREMENTAL:NO
+      /debug:full
+  )
 else ()
   # Windows handles exports with DLL_EXPORT defined above.
   set_target_properties(freetype PROPERTIES
@@ -515,7 +535,7 @@ endif ()
 target_include_directories(
   freetype
     PUBLIC
-      $<INSTALL_INTERFACE:include/freetype2>
+      $<INSTALL_INTERFACE:include>
       $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/include>
       $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
     PRIVATE
@@ -591,7 +611,7 @@ if (NOT SKIP_INSTALL_HEADERS AND NOT SKIP_INSTALL_ALL)
   install(
     # Note the trailing slash in the argument to `DIRECTORY'!
     DIRECTORY ${PROJECT_SOURCE_DIR}/include/
-      DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/freetype2
+      DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
       COMPONENT headers
       PATTERN "internal" EXCLUDE
       PATTERN "ftconfig.h" EXCLUDE
@@ -599,7 +619,7 @@ if (NOT SKIP_INSTALL_HEADERS AND NOT SKIP_INSTALL_ALL)
   install(
     FILES ${PROJECT_BINARY_DIR}/include/freetype/config/ftconfig.h
           ${PROJECT_BINARY_DIR}/include/freetype/config/ftoption.h
-      DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/freetype2/freetype/config
+      DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/freetype/config
       COMPONENT headers)
 endif ()
 
@@ -640,7 +660,7 @@ if (NOT SKIP_INSTALL_LIBRARIES AND NOT SKIP_INSTALL_ALL)
             FREETYPE2_PC_IN ${FREETYPE2_PC_IN})
   endif ()
 
-  set(FREETYPE2_PC_IN_NAME "${PROJECT_BINARY_DIR}/freetype2.pc")
+  set(FREETYPE2_PC_IN_NAME "${PROJECT_BINARY_DIR}/freetype.pc")
   if (EXISTS "${FREETYPE2_PC_IN_NAME}")
     file(READ "${FREETYPE2_PC_IN_NAME}" ORIGINAL_FREETYPE2_PC_IN)
   else ()
@@ -651,7 +671,7 @@ if (NOT SKIP_INSTALL_LIBRARIES AND NOT SKIP_INSTALL_ALL)
   endif ()
 
   install(
-    FILES ${PROJECT_BINARY_DIR}/freetype2.pc
+    FILES ${PROJECT_BINARY_DIR}/freetype.pc
     DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig
     COMPONENT pkgconfig)
 
@@ -662,22 +682,41 @@ if (NOT SKIP_INSTALL_LIBRARIES AND NOT SKIP_INSTALL_ALL)
     COMPATIBILITY SameMajorVersion)
 
   install(
-    TARGETS freetype freetype-interface
-      EXPORT freetype-targets
+    TARGETS freetype
+      EXPORT freetype
       LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
       ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
       RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
       FRAMEWORK DESTINATION Library/Frameworks
       COMPONENT libraries)
   install(
-    EXPORT freetype-targets
+    EXPORT freetype
       DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/freetype
-      FILE freetype-config.cmake
       COMPONENT headers)
+
+  install(FILES
+    cmake/freetype-config.cmake
+    ${CMAKE_CURRENT_BINARY_DIR}/freetype-config-version.cmake
+    DESTINATION lib/cmake/freetype)
+
+  install(
+    FILES ${PROJECT_BINARY_DIR}/RelWithDebInfo/freetype.pdb
+    DESTINATION lib
+    CONFIGURATIONS RelWithDebInfo
+  )
   install(
-    FILES ${PROJECT_BINARY_DIR}/freetype-config-version.cmake
-    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/freetype
-    COMPONENT headers)
+    FILES ${PROJECT_BINARY_DIR}/Debug/freetyped.pdb
+    DESTINATION lib
+    CONFIGURATIONS Debug
+  )
+  install(EXPORT ${PROJECT_NAME}
+  FILE
+    ${PROJECT_NAME}.cmake
+  NAMESPACE
+    ${PROJECT_NAME}::
+  DESTINATION
+    lib/cmake/${PROJECT_NAME}
+)
 endif ()
 
 
diff --git a/cmake/freetype-config.cmake b/cmake/freetype-config.cmake
new file mode 100644
index 000000000..1390d3ea2
--- /dev/null
+++ b/cmake/freetype-config.cmake
@@ -0,0 +1 @@
+include(${CMAKE_CURRENT_LIST_DIR}/freetype.cmake)
