diff --git a/CMakeLists.txt b/CMakeLists.txt
index 116c77dd..67d4b059 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -403,6 +403,8 @@ set_target_properties(
 )
 
 if(MSVC)
+    target_compile_definitions(LibXml2 PRIVATE _CRT_SECURE_CPP_OVERLOAD_STANDARD_NAMES)
+    target_compile_options(LibXml2 PRIVATE /MP /sdl-)
     if(BUILD_SHARED_LIBS)
         set_target_properties(
             LibXml2
