diff --git a/programs/fileio.c b/programs/fileio.c
index 77f3d35f..6188a8e2 100644
--- a/programs/fileio.c
+++ b/programs/fileio.c
@@ -799,6 +799,37 @@ static void FIO_munmap(FIO_Dict_t* dict)
     dict->dictBuffer = NULL;
     dict->dictBufferSize = 0;
 }
+#if defined(WINAPI_FAMILY) && WINAPI_FAMILY != WINAPI_FAMILY_DESKTOP_APP
+static inline wchar_t* to_utf16(const char* str)
+{
+    if (str == NULL)
+        return NULL;
+
+    size_t length = strlen(str);
+
+    int result = MultiByteToWideChar(CP_UTF8, 0, str, (int)length, NULL, 0);
+    if (result == 0)
+        return NULL;
+
+    length = (size_t)result + 1;
+    wchar_t* strW = (wchar_t*)malloc(length * sizeof(wchar_t));
+    if (strW == NULL)
+        return NULL;
+
+    result = MultiByteToWideChar(CP_UTF8, 0, str, result, strW, (int)length);
+
+    if (result == 0)
+    {
+        free(strW);
+        return NULL;
+    }
+
+    if (strW[length - 1] != L'\0')
+        strW[length - 1] = L'\0';
+
+    return strW;
+}
+#endif
 static size_t FIO_setDictBufferMMap(FIO_Dict_t* dict, const char* fileName, FIO_prefs_t* const prefs, stat_t* dictFileStat)
 {
     HANDLE fileHandle, mapping;
@@ -812,7 +843,11 @@ static size_t FIO_setDictBufferMMap(FIO_Dict_t* dict, const char* fileName, FIO_
 
     DISPLAYLEVEL(4,"Loading %s as dictionary \n", fileName);
 
+#if defined(WINAPI_FAMILY) && WINAPI_FAMILY != WINAPI_FAMILY_DESKTOP_APP
+    fileHandle = CreateFile2(to_utf16(fileName), GENERIC_READ, FILE_SHARE_READ, OPEN_EXISTING, NULL);
+#else
     fileHandle = CreateFileA(fileName, GENERIC_READ, FILE_SHARE_READ, NULL, OPEN_EXISTING, FILE_ATTRIBUTE_READONLY, NULL);
+#endif
 
     if (fileHandle == INVALID_HANDLE_VALUE) {
         EXM_THROW(33, "Couldn't open dictionary %s: %s", fileName, strerror(errno));
